client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -168,28 +169,28 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 				step: firstAuthWalledStep.slug,
 				lang: lang === 'en' || isLoggedIn ? null : lang,
 			} );
+
 			const signupUrl = generatePath( '/setup/:flow/:step/:lang?', {
 				flow: flow.name,
 				step: 'user',
 				lang: lang === 'en' || isLoggedIn ? null : lang,
 			} );
 
-			const lastPreAuthWalledStepIndex =
-				flowSteps.findIndex( ( step ) => step.slug === 'user' ) - 1;
-			const lastPreAuthWalledStep =
-				lastPreAuthWalledStepIndex < 0 ? null : flowSteps[ lastPreAuthWalledStepIndex ];
+			const flowStartsWithAuth = flowSteps[ 0 ].slug === 'user';
+			const hasHistory = location.key !== 'default';
 
 			return (
 				<StepComponent
 					navigation={ {
 						submit() {
 							navigate( firstAuthWalledStep.slug, undefined, true );
```

Comment:
It still made the flow become linear after the auth. The user will always be redirected to the first step that requires auth. However, there may be another path after the auth. For example, if the user comes from A2 step, we have to redirect them to B2 step after the auth. But the B1 step may be the first step that requires auth so that the user still goes to B2 step even if they are from the A2 step.

```
A1 -> auth -> B1
A2 -> auth -> B2
```